src/main/java/org/mitre/synthea/modules/covid/C19VaccineAgeDistributions.java

```diff
@@ -0,0 +1,233 @@
+    List<? extends Map<String, String>> rawRates = null;
+    try {
+      String csv = Utilities.readResource(fileName);
+      if (csv.startsWith("\uFEFF")) {
+        csv = csv.substring(1); // Removes BOM.
+      }
+      rawRates = SimpleCSV.parse(csv);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    rawRates.forEach((rowMap) -> {
+      String date = rowMap.get(DATE_COLUMN_HEADER);
+      rowMap.forEach((name, value) -> {
+        if (! name.equals(DATE_COLUMN_HEADER)) {
+          AgeRange r = new AgeRange(name);
+          List distroForRange = rawDistributions.get(r);
+          if (distroForRange == null) {
+            distroForRange = new ArrayList(200);
+            rawDistributions.put(r, distroForRange);
+          }
+          distroForRange.add(new Pair(date, Integer.parseInt(value)));
+        }
+      });
+    });
+    // Special case for 12-15. While there were individuals under 16 who received vaccinations
+    // prior to May 10, 2021, the numbers are pretty small. This strips out that population for
+    // the sake of simplicity in the simulation
+    AgeRange specialCaseRange = new AgeRange("Ages_12-15_yrs");
+    rawDistributions.put(specialCaseRange, rawDistributions.get(specialCaseRange).stream()
+        .filter(pair -> {
+          String dateString = pair.getFirst();
+          LocalDate doseCountDate = CSV_DATE_FORMAT.parse(dateString, LocalDate::from);
+          return doseCountDate.isAfter(EXPAND_AGE_TO_TWELVE);
+        })
+        .collect(Collectors.toList()));
+  }
+
+  /**
+   * Processes the raw distribution information into the EnumeratedDistributions that will be used
+   * to select the date of vaccination administration. Must be called after loadRawDistribution().
+   */
+  public static void populateDistributions() {
+    rawDistributions.forEach((ageRange, dayInfoList) -> {
+      double totalDosesForRange = dayInfoList.stream()
+          .map(pair -> pair.getSecond()).collect(Collectors.summingInt(Integer::intValue));
+      List<Pair<String, Double>> pmf = dayInfoList.stream().map(dayInfo -> {
+        double dosesForDay = dayInfo.getSecond();
+        double weight = dosesForDay / totalDosesForRange;
+        return new Pair<String, Double>(dayInfo.getFirst(), weight);
+      }).collect(Collectors.toList());
+      distributions.put(ageRange, new EnumeratedDistribution(pmf));
+    });
+  }
+
+  /**
+   * Load the JSON file that has a map of age ranges to percentage of that age range that has been
+   * vaccinated.
+   */
+  public static void loadShotProbabilitiesByAge() {
+    String fileName = "covid19/" + FIRST_SHOT_PROBS_FILE;
+    LinkedTreeMap<String, Object> rawShotProbs = null;
+    try {
+      String rawJson = Utilities.readResource(fileName);
+      Gson gson = new Gson();
+      rawShotProbs = gson.fromJson(rawJson, LinkedTreeMap.class);
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
You probably don't want to swallow this exception, because if you do, line 190 is just going to throw the dreaded null pointer exception.

Reply:
Now throws a `RuntimeException`